Aardvark/ARKEmailBugReporter.m

```diff
@@ -342,155 +336,128 @@ - (NSString *)formattedLogMessagesAttachmentExtension;
-    textField.returnKeyType = UIReturnKeyDone;
-}
-
-- (void)_createBugReportWithTitle:(NSString *)title;
-{
-    NSArray *logStores;
+    ARKEmailBugReportConfiguration *const configuration = [[ARKEmailBugReportConfiguration alloc] init];
+    
     if (self.emailAttachmentAdditionsDelegate != nil) {
         NSMutableArray *const filteredLogStores = [NSMutableArray arrayWithCapacity:self.logStores.count];
         for (ARKLogStore *logStore in self.logStores) {
             if ([self.emailAttachmentAdditionsDelegate emailBugReporter:self shouldIncludeLogStoreInBugReport:logStore]) {
                 [filteredLogStores addObject:logStore];
             }
         }
-        logStores = filteredLogStores;
+        configuration.logStores = filteredLogStores;
+        
+        configuration.additionalAttachments = [self.emailAttachmentAdditionsDelegate additionalEmailAttachmentsForEmailBugReporter:self];
+        
     } else {
-        logStores = [self.logStores copy];
+        configuration.logStores = [self.logStores copy];
     }
     
+    configuration.includesScreenshot = self.attachScreenshotToNextBugReport;
+    configuration.includesViewHierarchyDescription = (self.attachScreenshotToNextBugReport && self.attachesViewHierarchyDescriptionWithScreenshot);
+    
+    return configuration;
+}
+
+- (void)_createBugReportWithConfiguration:(ARKEmailBugReportConfiguration *)configuration;
+{
     NSMapTable *logStoresToLogMessagesMap = [NSMapTable new];
     NSDictionary *emailBodyAdditions = [self.emailBodyAdditionsDelegate emailBodyAdditionsForEmailBugReporter:self];
     
     if ([MFMailComposeViewController canSendMail]) {
         self.mailComposeViewController = [MFMailComposeViewController new];
         
         [self.mailComposeViewController setToRecipients:@[self.bugReportRecipientEmailAddress]];
-        [self.mailComposeViewController setSubject:title];
+        [self.mailComposeViewController setSubject:configuration.prefilledEmailSubject];
         
+        dispatch_group_t logStoreRetrievalDispatchGroup = dispatch_group_create();
+        dispatch_group_enter(logStoreRetrievalDispatchGroup);
+        
+        NSArray<ARKLogStore *> *const logStores = configuration.logStores;
         for (ARKLogStore *logStore in logStores) {
+            dispatch_group_enter(logStoreRetrievalDispatchGroup);
             [logStore retrieveAllLogMessagesWithCompletionHandler:^(NSArray *logMessages) {
                 [logStoresToLogMessagesMap setObject:logMessages forKey:logStore];
+                dispatch_group_leave(logStoreRetrievalDispatchGroup);
+            }];
+        }
+        
+        // Once all log messages have been retrieved, attach the data and show the compose window.
+        dispatch_group_notify(logStoreRetrievalDispatchGroup, dispatch_get_main_queue(), ^{
+            NSMutableString *emailBody = [self _prefilledEmailBodyWithEmailBodyAdditions:emailBodyAdditions];
```

Comment:
supernit: `const` here and... well... everywhere 😉

Reply:
Added it everywhere it seems relevant in this method.